internal/resolver/delegatingresolver/delegatingresolver_ext_test.go

```diff
@@ -202,6 +205,14 @@ func (s) TestDelegatingResolverwithDNSAndProxyWithTargetResolution(t *testing.T)
 	case <-time.After(defaultTestShortTimeout):
 	}
 
+	// Wait for the proxy resolver to be built before calling UpdateState.
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	select {
+	case <-proxyResolverBuilt:
+	case <-ctx.Done():
+		t.Fatalf("Context timed out waiting for proxy resolver to be built.")
+	}
```

Comment:
Another possible refactoring would be make this a function for this code block which is repeated in a lot of tests. Something like `func mustBuildResolverBuild(ctx context.Context, t *testing.T, buildCh chan struct{}) { ... }`.

See: go/go-style/decisions#must-functions